apps/iframe/src/Components/App/App.tsx

```diff
@@ -1,19 +1,132 @@
-import React from 'react'
+import React, { useState } from 'react'
+// @ts-ignore
+import * as hexToRgb from 'hex-to-rgb'
+
+import { ConnectService } from '../../services/connectService'
+import { UIProvider } from '@3id/ui-provider'
+import type { UIProviderHandlers } from '@3id/ui-provider'
+
 import './App.scss'
 import Modal from '../Modal/Modal'
 
 type AppProps = {
-  renderType?: string
-  bkgColor?: string
-
-  acceptFunction: any
-  declineFunction: any
+  // renderType?: string
+  // bkgColor?: string
+  // acceptFunction: any
+  // declineFunction: any
 }
 
-const App = ({ renderType, bkgColor }: AppProps) => {
+const App = () => {
+  const [renderFunctions, setRenderFunctions] = useState({
```

Comment:
Sorry if my previous comment wasn't clear about moving this logic to React. I didn't mean just moving the code inside a component but rather adapting the logic to be more in line with how React works.
It seems most of the logic here doesn't benefit from using React, notably in terms of separation of concerns between state and UIs.